subsys/net/l2/ethernet/gptp/gptp_messages.h

```diff
@@ -68,7 +68,7 @@ extern "C" {
 #define GPTP_ANNOUNCE_LEN(pkt) \
```

Comment:
The commit subject contains redundant info, you could shorten it to
```
net: gptp: Fix announce message len
```

Reply:
@jukkar, @mariopaja, this change is not correct (It breaks the announce message length in my gptp testing).

sizeof tlvType and sizeof lengthField are not taken into account in the value of lengthField. See below, the value of lengthField is 8*N, not 8*N + 4.

![image](https://github.com/zephyrproject-rtos/zephyr/assets/17433307/e1df0a21-9d05-472a-aff9-cc0d526a5f45)

Reply:
@talih0 Could you give a bit more context to the problem? Any error message, warning, or some Wireshark packets so we can check or have a look to the announce message? Which Zephyr version are you using? 

> sizeof tlvType and sizeof lengthField are not taken into account in the value of lengthField. See below, the value of lengthField is 8_N, not 8_N + 4.

Hmmm I did not understand your point, because it doesn't write anywhere that **lengthField is 8_N**. 
Based on the standard:
tlvType = 2 bytes
lengthField = 2 bytes
pathSequence = 8N bytes (Zephyr considers only 8 bytes)


BTW the document you are using is an unapproved Draft and at **10.6.3.3.2** it says _"Table 50 of IEEE Std 1588-2019"_. I don't know what is written in that standard but as I know all the MCU that I have worked until now support only IEEE Std 1588-2008, there might also be significant differences.

Reply:
> @talih0 Could you give a bit more context to the problem? Any error message, warning, or some Wireshark packets so we can check or have a look to the announce message? Which Zephyr version are you using?
> 
> > sizeof tlvType and sizeof lengthField are not taken into account in the value of lengthField. See below, the value of lengthField is 8_N, not 8_N + 4.
> 
> Hmmm I did not understand your point, because it doesn't write anywhere that **lengthField is 8_N**. 

That's what it says in the figure I pasted above :)  See 10.6.3.3.3

It's also how Anvu/gptp encodes the announce length field:
https://github.com/Avnu/gptp/blob/master/common/avbts_message.hpp#L475

Reply:
and confirming with wireshark when using Avnu/gptp:

![image](https://github.com/zephyrproject-rtos/zephyr/assets/17433307/2d7d00a9-87d0-4a78-9a0b-942779efbadb)

Reply:
@talih0 I think you have misunderstood 10.6.3.3.3 and the Path Trace TLV table. 

Path Trace TLV table explains the size in Octets (bytes) of each component that belongs in it (tlvType, lengthField, pathSequence)

10.6.3.3.3 says **The value of lengthField is 8N**. Therefore, lengthField can be any 8N value as long as it is 2 bytes.

Reply:
@mariopaja Please double-check on your side again. 

lengthField = sizeof(valueField). It is NOT sizeof(valueField) + sizeof(tlvType) + sizeof(lengthField)

https://www.shoshin.co.jp/c/endrun/1588ptp/IEEE1588v1_vs_IEEE1588v2.pdf
![image](https://github.com/zephyrproject-rtos/zephyr/assets/17433307/0137d6a1-4be4-46cd-9cb3-10105b8fad58)

Reply:
@talih0 you are correct on this fact.
> lengthField = sizeof(valueField)

**lengthField** value is the number of bytes that are on the valueField, in this case 8. **lengthField** size 2 bytes. 

Path Trace TLV = 12 bytes
<img width="683" alt="Screenshot 2023-12-16 at 18 05 07" src="https://github.com/zephyrproject-rtos/zephyr/assets/28357816/f01ca609-6c35-4376-b77d-8218707355bc">

tlvType = 2 bytes
<img width="667" alt="Screenshot 2023-12-16 at 18 05 20" src="https://github.com/zephyrproject-rtos/zephyr/assets/28357816/0ac34f50-fced-46d4-adb4-80ff966d2198">

lengthField = 2 bytes
<img width="665" alt="Screenshot 2023-12-16 at 18 05 29" src="https://github.com/zephyrproject-rtos/zephyr/assets/28357816/b3655593-9073-48bc-93db-60d0fbd5d4ce">

pathSequence = 8 bytes
<img width="652" alt="Screenshot 2023-12-16 at 18 05 37" src="https://github.com/zephyrproject-rtos/zephyr/assets/28357816/5fa82d4c-71d1-4b03-9633-0671c0d6fd2e">

_images taken from the Wireshark log attached to the PR_

Reply:
@mariopaja, the +4 that was removed in this PR is needed for the tlvType and lengthField (each is 2 bytes). I think I have been clear enough. 

@jukkar, @rlubos, please have another look. The Announce messages are not parsed properly when Avnu/gptp (https://github.com/Avnu/gptp) is used for the master in the gptp sample.

Reply:
@talih0 I am trying to understand your point but until now you have provided only fragments of your problem  without clear argumentation. Could you send a full Wireshark log to see the full announce messages?

What is your setup? What MCU are you using? Are your devices connected directly with each other or is there any managed switch in the middle?

Reply:
> @talih0 I am trying to understand your point but until now you have provided only fragments of your problem without clear argumentation.

@mariopaja, I appreciate that you want to have a look at the logs. 

I'm using beaglebone black running Avnu/gptp connected to xmc47_relax_kit directly (the ethernet drivers are not upstreamed yet, but it should be MCU agnostic on the zephyr stack).

With your patch I always get:
[00:00:21.683,000] <wrn> net_gptp: Invalid length for ANNOUNCE packet should have 72 bytes but has 76 bytes

I put a switch in the middle to capture the pcap file:
[xmc47_beagleboard.pcap.zip](https://github.com/zephyrproject-rtos/zephyr/files/13694099/xmc47_beagleboard.pcap.zip)

Reply:
@mariopaja, I also had a look at the pcap file you attached in the PR. All the announce messages appear to have a size of 76 bytes consistent with the above error message.

Reply:
@talih0 It is interesting. I think we might both be right in our setup configuration and maybe this driver should be further developed beyond the **+4** case to accommodate both our cases. 

We are using a totally different setup. We are using 1 x RPi CM4 running _**LinuxPTP**_, 1 x Nucleo H745Zi-Q and 2 x custom MIMXRT1064. All devices are connected with a managed switch that supports gPTP and the devices do not have the same priority prio1. Usually we keep the Nucleo board with the best master priority (prio1).

In our case we noticed that the switch does a lot of the heavy lifting and it does not forward the TLV on the announce message. See below:
[gPTP-Slave-Ethertap-log291123.pcapng.zip](https://github.com/zephyrproject-rtos/zephyr/files/13695384/gPTP-Slave-Ethertap-log291123.pcapng.zip)

Reply:
@mariopaja, yes, this needs to fixed properly. And I think it's only by chance that your setup is working. The tlv length is still being dereferenced now even though the segment doesn't exist:
https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/net/l2/ethernet/gptp/gptp_messages.h#L70

Reply:
@fabiobaltieri @jukkar  Maybe we can revert this PR since it causes problems for @talih0 and perhaps other developers too.

Maybe we can think of another possibility to address my case where there is no TLV on the Announce message, since it is optional based on 10.5.1 IEEE Std. 802.1AS-2011

![10 5 1](https://github.com/zephyrproject-rtos/zephyr/assets/28357816/fdd5f0a9-6ea6-4adf-a66f-db0723246b53)

Reply:
> Maybe we can revert this PR since it causes problems for @talih0 and perhaps other developers too.

Please send a PR that reverts it. Add also some info in the commit message why the commit is reverted.